back/src/Guide/Cli.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
+-- | This module adds command line interface to run server.
+--
+-- See help with 'guide --help' to see available commands.
+module Guide.Cli
+       ( Command (..)
+       , parseCli
+       ) where
+
+
+import Imports
+
+import Data.Version (showVersion)
+import Development.GitRev (gitCommitDate, gitDirty, gitHash)
+import Options.Applicative (Parser, ParserInfo, command, execParser, fullDesc, help, helper, info,
+                            infoHeader, infoOption, long, metavar, progDesc, short, strOption,
+                            subparser, (<|>))
+import Options.Applicative.Help.Chunk (stringChunk)
+
+import Paths_guide (version)
+
+import qualified Data.Text as T
+
+
+-- | All available commands
+data Command
+    = Guide               -- ^ run server
+    | DryRun              -- ^ load database and exit
+    | LoadPublic String   -- ^ load PublicDB, create base on it and exit
+
+----------------------------------------------------------------------------
+-- Parsers
+----------------------------------------------------------------------------
+
+-- | Main parser of the application.
+parseCli :: IO Command
+parseCli = execParser
+    $ modifyHeader
+    $ info (helper <*> versionP <*> (pure Guide <|> commandsP))
+    $ fullDesc -- <> progDesc "Guide is wiki"
+
+-- | All possible commands.
+commandsP :: Parser Command
+commandsP = subparser
+    $  command "run" (info (helper <*> pure Guide) $ progDesc "Run server")
+    <> command "dry-run" (info (helper <*> pure DryRun) $ progDesc "Load database and exit")
+    <> command "load-public" (info (helper <*> loadPublic) $ progDesc "Load PublicDB, create base on it and exit")
```

Comment:
Please try to keep lines shorter than 80 chars, otherwise working with a split screen is hard.